components/event-bus/cmd/event-publish-service/application/application.go

```diff
@@ -56,7 +57,10 @@ func (app *KnativePublishApplication) start() {
 
 	// mark the app as started and register the readiness and the publish handlers
 	app.started = true
-	app.registerReadinessProbe()
+	if app.knativeLib != nil {
```

Comment:
Shouldn't we just let it panic if nil? Doesn't sound like something we want to recover from.

Reply:
Alright, added a panic.